tests/entities/repos/test_submission_repo.py

```diff
@@ -193,20 +193,39 @@ async def test_add_filing_task(self, query_session: AsyncSession, transaction_se
             seconds_now, abs=1.0
         )  # allow for possible 1 second difference
 
-    async def test_get_filing(self, query_session: AsyncSession):
-        res = await repo.get_filing(query_session, lei="1234567890", filing_period="2024")
-        assert res.id == 1
-        assert res.filing_period == "2024"
-        assert res.lei == "1234567890"
-        assert len(res.tasks) == 2
-        assert FilingTaskState.NOT_STARTED in set([t.state for t in res.tasks])
-
-        res = await repo.get_filing(query_session, lei="ABCDEFGHIJ", filing_period="2024")
-        assert res.id == 2
-        assert res.filing_period == "2024"
-        assert res.lei == "ABCDEFGHIJ"
-        assert len(res.tasks) == 2
-        assert FilingTaskState.NOT_STARTED in set([t.state for t in res.tasks])
+    async def test_get_filing(self, query_session: AsyncSession, mocker: MockerFixture):
+        spy_populate_missing_tasks = mocker.patch(
+            "entities.repos.submission_repo.populate_missing_tasks", wraps=repo.populate_missing_tasks
+        )
+        res1 = await repo.get_filing(query_session, lei="1234567890", filing_period="2024")
+        assert res1.id == 1
+        assert res1.filing_period == "2024"
+        assert res1.lei == "1234567890"
+        assert len(res1.tasks) == 2
+        assert FilingTaskState.NOT_STARTED in set([t.state for t in res1.tasks])
+        tasks1 = set([task_state.task for task_state in res1.tasks if task_state.task])
```

Comment:
Are the if's needed here?  With your update, shouldn't be a None to check for should there?

Also, is it just me or did I name these DAO/DTOs poorly, now that I read it?  FilingTaskState, FilingTask, etc.  Just doesn't read well, even I get confused what objects are being held where and I fricking wrote it.

Reply:
ah yeap, if's not needed here, since if it's none, then the name wouldn't get populated either.
lol yeah, I'll come up with something more differentiating.